src/llvm/tools/clang/lib/Driver/ToolChains/MSP430.cpp

```diff
@@ -24,6 +24,22 @@ using namespace clang::driver::tools;
 using namespace clang;
 using namespace llvm::opt;
 
+static bool isSupportedMCU(const StringRef MCU) {
+  return llvm::StringSwitch<bool>(MCU)
+#define MSP430_MCU(NAME) .Case(NAME, true)
+#include "clang/Basic/MSP430Target.def"
+      .Default(false);
+}
+
+void msp430::getMSP430TargetFeatures(const Driver &D, const ArgList &Args,
+                                     std::vector<StringRef> &Features) {
+  const Arg *MCUArg = Args.getLastArg(options::OPT_mmcu_EQ);
+  if (MCUArg && !isSupportedMCU(MCUArg->getValue())) {
+    D.Diag(diag::err_drv_msp430_unknown_mcu) << MCUArg->getValue();
```

Comment:
It's better to report warning here and proceed to build in "compatibility" mode for msp430x and msp430xv2 devices.